Tests/SonarScanner.MSBuild.Shim.Test/SonarScannerWrapperTests.cs

```diff
@@ -257,48 +267,72 @@ public void SonarScanner_UserSpecifiedEnvVars_OnlySONARSCANNEROPTSIsPassed()
+
+        [TestMethod]
+        public void FindScannerExe_FindsCliZip_ExtractsItSuccessfully()
+        {
+            // Arrange
+            var logger = new TestLogger();
+
+            var scannerCliTestDirectory = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext, new string[] { "sonar-scanner-cli-0", "sonar-scanner-0", "bin" });
+            TestUtils.CreateEmptyFile(scannerCliTestDirectory, "sonar-scanner.bat");
+
+            var scannerCliTestDirectoryRoot = Directory.GetParent(scannerCliTestDirectory).Parent.FullName;
+            var zipDestinationDir = Path.Combine(TestContext.TestDir, "sonar-scanner-cli-0.zip");
+            ZipFile.CreateFromDirectory(scannerCliTestDirectoryRoot, zipDestinationDir);
+
+            // Act
+            SonarScannerWrapper.FindScannerExe(logger, TestContext.TestDir, "0");
+
+            // Assert
+            Directory.Exists(Path.Combine(TestContext.TestDir, "sonar-scanner-0"));
+            logger.AssertInfoMessageExists("Unzipping sonar-scanner-cli-0.zip");
         }
 
-        private void TestWrapperErrorHandling(bool executeResult, bool addMessageToStdErr, bool expectedOutcome)
+        [TestMethod]
+        public void FindScannerExe_ZipIsAlreadyExtracted_DoesNotExtractAgain()
         {
             // Arrange
             var logger = new TestLogger();
-            var mockRunner = new MockProcessRunner(executeResult);
+            var scannerCliTestDirectory = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext, new string[] { "sonar-scanner-0", "bin" });
+            TestUtils.CreateEmptyFile(scannerCliTestDirectory, "sonar-scanner.bat");
+            var scannerCliTestDirectoryRoot = Directory.GetParent(scannerCliTestDirectory).Parent;
 
-            var config = new AnalysisConfig() { SonarScannerWorkingDirectory = "C:\\working" };
+            // Act
+            SonarScannerWrapper.FindScannerExe(logger, scannerCliTestDirectoryRoot.FullName, "0");
 
-            if (addMessageToStdErr)
-            {
-                logger.LogError("Dummy error");
-            }
+            // Assert
+            logger.AssertMessageNotLogged("Unzipping sonar-scanner-cli-0.zip");
+        }
+
+        [TestMethod]
+        public void FindScannerExe_CannotFindScannerCliZip_Throws()
+        {
+            // Arrange
+            var logger = new TestLogger();
+            var testDirectory = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext);
 
             // Act
-            var success = ExecuteJavaRunnerIgnoringAsserts(config, Enumerable.Empty<string>(), logger, "c:\\bar.exe", "c:\\props.xml", mockRunner);
+            Action act = () => SonarScannerWrapper.FindScannerExe(logger, testDirectory, "0");
 
             // Assert
-            VerifyProcessRunOutcome(mockRunner, logger, "C:\\working", success, expectedOutcome);
+            act.Should().ThrowExactly<FileNotFoundException>();
```

Comment:
Will the `FileNotFoundException` thrown by `ExtractToDirectory` end up in the logs? In that case I am not sure it is worth it checking for the path existence and do nothing. I would either do the check and throw an exception or do nothing and let the `ExtractToDirectory` method handle the case. @andrei-epure-sonarsource  WDYT about it?

Reply:
Yeah, the exception will also contain the message that will end up in the user's log.
Should be safe to delete the chack.

Reply:
The check has been deleted. The exception message from `ExtractToDirectory ` is meaningful enought to stand on its own.